src/matUtils/describe.cpp

```diff
@@ -1,102 +1,115 @@
+                // }
+                if (n != root_to_sample.back()) {
+                    //note, for uncondensed samples with parsimony score 0,
+                    //this will leave a > at the end. That's basically fine.
+                    cpath += " > ";
+                    //fprintf(stdout, " > ");
+                }
+            }
+        }
+        mpaths.push_back(cpath);
+        //fprintf(stdout, "\n");
+    }
+    return mpaths;
+}
+
+std::vector<std::string> clade_paths(MAT::Tree T, std::vector<std::string> clades) {
+    //get the set of clade path strings for printing
+    //similar to the above, construct a series of strings to be printed or redirected later on
+    std::vector<std::string> clpaths;
+    //NOTE: setting aside functionality of getting unique mutations for clarity for the moment, may reimplement later.
+    clpaths.push_back("clade\troot_id\tfrom_tree_root");
+    //clpaths.push_back("clade\troot_id\tspecific\tfrom_tree_root");
+    //do a breadth-first search
+    //the first time a clade that is in clades is encountered, that's the root;
+    //get the path of mutations to that root (rsearch), save the unique mutations + that path
+    //unique mutations being ones that occurred in the clade root, and the path being all mutations from that root back to the tree root
+    //then continue. if a clade has already been encountered in the breadth first, its
+    //not clade root, and should be skipped.
+    std::unordered_set<std::string> clades_seen;
+
+    auto dfs = T.breadth_first_expansion();
+    for (auto n: dfs) {
+        std::string curpath;
+        for (auto ann: n->clade_annotations) {
+            if (ann != "") {
+                //if its one of our target clades and it hasn't been seen...
+                if (std::find(clades.begin(), clades.end(), ann) != clades.end() && clades_seen.find(ann) == clades_seen.end()) {
+                    //record the name of the clade
+                    curpath += ann + "\t";
+                    curpath += n->identifier + "\t";
+                    // //get its own mutations, if there are any
+                    // std::string unique = "";
+                    // for (size_t i=0; i<n->mutations.size(); i++) {
+                    //     unique += n->mutations[i].get_string();
+                    //     if (i+1 < n->mutations.size()) {
+                    //         unique += ",";
+                    //     }
+                    // }
+                    // curpath += unique + "\t";
+                    //get the ancestral mutations back to the root
+                    std::string root = "";
+                    auto root_to_sample = T.rsearch(n->identifier, true);
+                    std::reverse(root_to_sample.begin(), root_to_sample.end());
+                    for (auto an: root_to_sample) {
+                        //skip the root.
+                        if (!an->is_root()) {

```

Comment:
Not always, but root can also have mutations

Reply:
ah the root in my test PB had no mutations and it seemed useless. I can remove the check